site/content/blog/making-the-most-out-of-clis.md

```diff
@@ -0,0 +1,93 @@
+Learn more about these commands [here](https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#-strong-cluster-management-strong-).
+
+### Dive into your containers and applications
+`kubectl` allows us to attach to processes containers running on pods and execute commands inside them easily. Users can observe the resource consumption on a pod as well.
+// might remove
+Proxying and port-forwarding options let a pod or the K8s API server serve HTTP content to localhost. Letting users debug their applications easily
+
+## `kapp`
+`kapp` is a CLI that helps users manage their workloads effectively on the cluster. This involves defining interactions between resources declaratively, deploying resources reliably and updating them safely and confidently.
+
+But wait isn't this something `kubectl apply` or `kubectl patch` will help us do for resource manifests?
+
+Well yes, but `kapp` goes a step further, it helps users order resources in their manifests, apply changes safely and predictably, and then goes on to wait for the resources to reach their desired state. (You could take a deep dive into this over [here](https://carvel.dev/kapp/docs/latest/))
+
+Let's see how `kapp` makes our lives easier.
+
+### Grouping related resources
+`kapp` lets users group and [order multiple resources](https://carvel.dev/kapp/docs/latest/apply-ordering/#example) easily. This lets us do tasks like "create a Deployment after a Job has finished" declaratively in our manifests.
+`kapp` also caters to other use cases, like [versioning resources](https://carvel.dev/kapp/docs/latest/diff/#versioned-resources) and defining [create/update/delete strategies](https://carvel.dev/kapp/docs/latest/apply/#controlling-apply-via-resource-annotations) for resources in the abstract "application". These features allow users to efficiently and declaratively manage resources to be deployed on clusters.
+
+### Applying changes safely and confidently
+While applying changes to the cluster, `kapp` shows a diff against the resources on the cluster (as seen [here](https://carvel.dev/kapp/docs/latest/)), this lets the user be more aware of the changes a patch or a deploy has on the cluster.
+Once the user confirms the changes, `kapp` waits for resources to reach their desired state ensuring that the changes are in effect as the user intended.
+
+### View the grouped resources and their statuses easily
+The `kapp inspect` command enables users to view resources that have been grouped into an "application". It lists the resources and surfaces information about whether or not the resources are in a healthy state. In case a resource is not in it's desired state, helpful information which helps the user work towards the cause of the failure is surfaced as well.
```

Comment:
nit: `kapp inspect` instead of `The kapp inspect`

Reply:
I think having an article here makes sense 🤔